core/execution/market_snapshot.go

```diff
@@ -155,7 +155,7 @@ func NewMarketFromSnapshot(
 		marketActivityTracker:      marketActivityTracker,
 		positionFactor:             positionFactor,
 		stateVarEngine:             stateVarEngine,
-		settlementDataInMarket:     em.SettlementData,
+		settlementDataInMarket:     &em.SettlementData,
```

Comment:
This doesn't quite make sense. If a field can be a nil pointer in the market, but not on the `ExecMarket`/snapshot type, then restoring a snapshot arguably doesn't restore the _exact_ same state. I believe the snapshot hash would still match in this case, but it doesn't look entirely clean IMO

Reply:
Right, set it to be a pointer in both places. Thanks!